lib/fog/openstack/models/identity_v3/projects.rb

```diff
@@ -8,8 +8,20 @@ class V3
         class Projects < Fog::OpenStack::Collection
           model Fog::Identity::OpenStack::V3::Project
 
+          @@cache = {}
+          Fog::Identity::OpenStack::V3::Project.use_cache(@@cache)
+
           def all(options = {})
-            load_response(service.list_projects(options), 'projects')
+            cached_project, expires = @@cache[{token: service.auth_token, options: options}]
```

Comment:
@dhague ah, so the reason is I was calling just .projects method, in this case, it is caching empty list for me

It works when I call .projects.all.

So seems like there is some lazy evaluation in place? That is calling .all method later?